django/db/migrations/loader.py

```diff
@@ -315,6 +319,32 @@ def _reraise_missing_dependency(migration, missing, exc):
                         # Since we added "key" to the nodes before this implies
                         # "child" is not in there.
                         _reraise_missing_dependency(migration, child, e)
+        # Check that no migrations are applied before their dependencies
```

Comment:
Maybe, it would be better to call this method from `migrate` and `makemigrations` commands? In this way, it will not be called on `showmigration`.
Also, it could be `check_consistent_history(self, connection)` and in `makemigrations` we can check all database connections.